src/sinks/util/tls.rs

```diff
@@ -93,59 +97,65 @@ impl TryFrom<&TlsOptions> for TlsSettings {
-                    options.key_path.as_ref().unwrap(),
-                    &options.key_pass,
-                    crt_path,
-                )?;
-                Some(
-                    Identity::from_pkcs12(&identity.to_der().context(DerExportError)?, "")
-                        .context(TlsIdentityError)?,
-                )
+                let name = crt_path.to_string_lossy().to_string();
+                let crt = load_x509(crt_path)?;
+                let key_path = options.key_path.as_ref().unwrap();
+                let key = load_key(&key_path, &options.key_pass)?;
+                let pkcs12 = Pkcs12::builder()
+                    .build("", &name, &key, &crt)
+                    .context(Pkcs12Error)?;
+                let identity = pkcs12.to_der().context(DerExportError)?;
+
+                // Build the resulting Identity, but don't store it, as
+                // it cannot be cloned.  This is just for error
+                // checking.
+                let _identity = Identity::from_pkcs12(&identity, "").context(TlsIdentityError)?;
+
+                Some(identity)
             }
         };
 
         Ok(Self {
-            verify_certificate: options.verify_certificate.unwrap_or(true),
+            accept_invalid_certificates: !options.verify_certificate.unwrap_or(true),
+            accept_invalid_hostnames: !options.verify_hostname.unwrap_or(true),
             authority,
             identity,
         })
     }
 }
 
-impl TryFrom<TlsOptions> for TlsSettings {
-    type Error = crate::Error;
-    fn try_from(options: TlsOptions) -> Result<Self, Self::Error> {
-        Self::try_from(&options)
-    }
-}
-
 pub trait TlsConnectorExt {
     fn use_tls_settings(&mut self, settings: TlsSettings) -> &mut Self;
 }
 
 impl TlsConnectorExt for TlsConnectorBuilder {
     fn use_tls_settings(&mut self, settings: TlsSettings) -> &mut Self {
-        self.danger_accept_invalid_certs(!settings.verify_certificate);
+        self.danger_accept_invalid_certs(settings.accept_invalid_certificates);
+        self.danger_accept_invalid_hostnames(settings.accept_invalid_hostnames);
         if let Some(certificate) = settings.authority {
             self.add_root_certificate(certificate);
         }
         if let Some(identity) = settings.identity {
+            // This data was test-built previously, so we can just use
+            // it here and expect the results will not fail. This can
+            // all be reworked when `native_tls::Identity` gains the
+            // Clone impl.
+            let identity = Identity::from_pkcs12(&identity, "").expect("Could not build identity");
             self.identity(identity);
         }
         self
     }
 }
 
 /// Load a `native_tls::Certificate` from a named file
-pub fn load_certificate<T: AsRef<Path> + Debug>(filename: T) -> crate::Result<Certificate> {
+fn load_certificate<T: AsRef<Path> + Debug>(filename: T) -> crate::Result<Certificate> {
```

Comment:
not blocking and take it or leave it, I like using impl Trait here actually find it creates a bit more readable code.
```suggestion
fn load_certificate(filename: impl AsRef<Path> + Debug) -> crate::Result<Certificate> {
```

Reply:
I agree `impl Trait` makes the function signature visibly simpler, but it may also mask that it's a generic function as opposed to something that takes a trait object (`dyn Trait`). I'm on the fence which is more readable.

I could always eliminate the problem by just making it take `&Path` since all the callers are local anyways.